flake.nix

```diff
@@ -0,0 +1,87 @@
+{
+  description = "Gnome monitor config CLI tool";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+  };
+
+  outputs = { self, nixpkgs }:
+    let
+      binName = "gnome-monitor-config";
+      supportedSystems = [ "x86_64-linux" "aarch64-linux" ];
+      forAllSystems = function:
+        nixpkgs.lib.genAttrs supportedSystems (system:
+          function { 
+            inherit system;
+            pkgs = import nixpkgs { inherit system; };
+          }
+        );
+    in {
+      packages = forAllSystems ({ pkgs, ... }:
+        {
+          default = pkgs.clangStdenv.mkDerivation {
+            name = binName;
+            src = ./.;
+
+            buildInputs = with pkgs; [
+              cairo
+              cmake
+              meson
+              ninja
+              pkg-config
+            ];
+
+            configurePhase = ''
+              meson build
+            '';
+
+            buildPhase = ''
+              (
+                cd build
+                meson compile
+              )
+            '';
+
+            installPhase = ''
+              mkdir -p $out/bin
+              cp build/src/${binName} $out/bin
+            '';
```

Comment:
Don't use `name` use `pname`.
There's build time tools in `buildInputs`.
The phases are unneeded because of meson's hook.
It would make more sense to `override` the stdenv to clang and  `overrideAttrs` the src of the package from nixpkgs.

Reply:
Thanks for the feedback. I've switched to `pname` and `nativeBuildInputs`.

I'm a flake noob. Could you elaborate on the override suggestion or point me to an example? Are you suggesting I override from an existing package in nixpkgs?